Problem should rather be fixed in jpeglib.h on Solaris where
jpeglib.h does not behave as "system library header" from the
C++ point of view because the system headers should define
C++ compatibility internally.

Not suitable for community.

--- qpdf-release-qpdf-10.3.2/include/qpdf/Pl_DCT.hh.orig	2021-08-27 16:00:30.192717127 +0200
+++ qpdf-release-qpdf-10.3.2/include/qpdf/Pl_DCT.hh	2021-08-27 16:01:41.526774361 +0200
@@ -24,7 +24,9 @@
 
 #include <qpdf/Pipeline.hh>
 #include <qpdf/Pl_Buffer.hh>
+extern "C" {
 #include <jpeglib.h>
+}
 
 class Pl_DCT: public Pipeline
 {
